### ISSUE TITLE
Use contextvars for the nursery fixture

### ISSUE THREAD

**Issue by touilleMan** [python-trio/pytest-trio#54]
Right now the nursery fixture is implemented in a pretty hacky way by patching the item object given by pytest:
https://github.com/python-trio/pytest-trio/blob/a972dc614826d7e2b24da356a4bc2dbffaf39098/pytest_trio/plugin.py#L282-L284

I wonder if it wouldn't be better to use the contextvars feature now that it is availble (even in older version of python thanks to Yury's backport https://github.com/MagicStack/contextvars)

**Comment by njsmith** [comment]
No longer relevant after #50 made the nursery fixture unique for each fixture/test that requests it.
